### ISSUE TITLE
[Feature request] Profiles for menubar layout

### ISSUE THREAD

**Issue by rebdeg** [jordanbaird/Ice#26]
Profiles with specific settings/layout of the menubar for different tasks (work, personal) etc would be a nice feature

**Comment by jordanbaird** [comment]
Good idea! It will have to wait until individual item movement is implemented, but I'll add this to the roadmap for the time being

**Comment by Xytronix** [comment]
For the layout part of the app this is an interesting feature. Assigning to different focus modes.

**Comment by jordanbaird** [comment]
@Xytronix love this idea. Would really make it feel like part of macOS.

**Comment by llity** [comment]
Profiles can be synchronized via iCloud.

**Comment by JaKXz** [comment]
I was going to write a new request for this: I've noticed that when I quit and re-open Ice for whatever reason the items I've put in the hidden and always hidden sections get reset/randomized somehow. I would like if Ice remembered how I had organized the sections with apps.

I'm happy to write a new issue for that if it's preferred, but I thought this would be a helpful +1.

**Comment by dsully** [comment]
It'd also be great to be able to automatically switch profiles based on SSID or other variables.

**Comment by stillworks** [comment]
I'm getting into focus modes and some apps that I'd like to move to the visible area for videocalls and other things.
Any idea when to expect that feature?

**Comment by partrap** [comment]
I would also love profiles! I'm big fan of Ice and it would be a great feature for me now that I have a couple long menu items with verbose labels (timer intentions, upcoming calendar events etc.).

**Comment by acousticbiker** [comment]
Great idea for a great app! Would be great for using full menubar space when connected to monitor vs smaller space when traveling and using MacBook Air display (and want to see different things like time at home)

**Comment by acousticbiker** [comment]
> It'd also be great to be able to automatically switch profiles based on SSID or other variables.

One helpful example would be when plugged into a monitor (allowing for much more menubar space)

**Comment by philocalyst** [comment]
If profiles are launched, I wonder if support would be considered for a declarative configuration file for them? Something in TOML/YAML/Etc, that users could carry around from machine to machine? It might mean that an menu-bar inspector tool would have to be created (although this could maybe just be an expansion of tooltips to include their accessibility element ID alongside a bundle identifier). This is just an expansion of the ideas proposed in #326 though!

**Comment by KristijanZic** [comment]
Please try to make these profiles portable via configuration files somewhere in ./config
Whatever changes are configured in the app should be reflected in config file and whatever changes are made in the config file should be reflected in the app.

**Comment by quintrino** [comment]
Yeah, having a declarative configuration file that lives in $XDG_CONFIG_HOME / would be amazing.

**Comment by comatory** [comment]
Second this, I actually backup my stuff via `yadm` and every software that supports a config file makes this very easy. Having it in $XDG_CONFIG_HOME would help me a lot.
